akka-actor-tests/src/test/scala/akka/event/EventStreamSpec.scala

```diff
@@ -80,11 +80,18 @@ class EventStreamSpec extends AkkaSpec(EventStreamSpec.config) {
       //#event-bus-start-unsubscriber-scala
 
       bus.subscribe(testActor, classOf[M])
-      bus.publish(M(42))
-      within(1.second) {
-        expectMsg(M(42))
-        bus.unsubscribe(testActor)
-        bus.publish(M(13))
+      // subscription is async
+      var n = 41
+      awaitAssert {
+        n += 1
+        bus.publish(M(n))
+        expectMsg(M(n))
```

Comment:
Not sure we actually need a unique message here per try, but it probably doesn't hurt either.

Reply:
isn't the timeout 3 seconds for both the expectMsg and awaitAssert? if the testActor hasn't been subscribed yet when publishing M(42) there will be a timeout on `expectMsg(42)`. Shouldn't we have a shorter timeout on expectMsg so that the `bus.publish` is retried?

Too short timeout on expectMsg  is also somewhat dangerous because then M(42) could be in flight and we retry with publishing M(43), then receiving M(42) and so on.

Reply:
maybe it could use something like fishForMessage?

Reply:
Also, looking at this again, I think I misread subscriptions, they are not async, what is async is the registration with the unsubscriber, so the original logic was ok, I'll revert.